qa/python_models/custom_metrics/model.py

```diff
@@ -128,14 +218,56 @@ def test_gauge_e2e(self):
             description="test metric gauge kind end to end",
             kind=pb_utils.MetricFamily.GAUGE,
         )
-        labels = {"example1": "counter_label1", "example2": "counter_label2"}
+        labels = {"example1": "gauge_label1", "example2": "gauge_label2"}
         metric = metric_family.Metric(labels=labels)
         self._metric_api_helper(metric, "gauge")
 
-        pattern = 'test_gauge_e2e{example1="counter_label1",example2="counter_label2"}'
+        pattern = 'test_gauge_e2e{example1="gauge_label1",example2="gauge_label2"}'
         metrics = self._get_metrics()
         self.assertIn(pattern, metrics)
 
+    def test_histogram_e2e(self):
+        name = "test_histogram_e2e"
+        metric_family = pb_utils.MetricFamily(
+            name=name,
+            description="test metric histogram kind end to end",
+            kind=pb_utils.MetricFamily.HISTOGRAM,
+        )
+
+        labels = {"example1": "histogram_label1", "example2": "histogram_label2"}
+        buckets = [0.1, 1.0, 2.5, 5.0, 10.0]
+        metric = metric_family.Metric(labels=labels, buckets=buckets)
+
+        labels_str = 'example1="histogram_label1",example2="histogram_label2"'
+        self._histogram_api_helper(metric, name, labels_str)
+
+        metrics = self._get_metrics()
+        count_pattern = f"{name}_count{{{labels_str}}}"
+        sum_pattern = f"{name}_sum{{{labels_str}}}"
+        bucket_pattern = f"{name}_bucket{{{labels_str}"
+        self.assertEqual(metrics.count(count_pattern), 1)
+        self.assertEqual(metrics.count(sum_pattern), 1)
+        self.assertEqual(metrics.count(bucket_pattern), len(buckets) + 1)
+
+    def test_histogram_args(self):
+        name = "test_histogram_args"
+        metric_family = pb_utils.MetricFamily(
+            name=name,
+            description="test metric histogram args",
+            kind=pb_utils.MetricFamily.HISTOGRAM,
+        )
+
+        # Test non-ascending order buckets
+        with self.assertRaises(pb_utils.TritonModelException):
+            metric_family.Metric(labels={}, buckets=[2.5, 0.1, 1.0, 10.0, 5.0])
+
+        # Test duplicate value buckets
+        with self.assertRaises(pb_utils.TritonModelException):
+            metric_family.Metric(labels={}, buckets=[1, 1, 2, 5, 5])
+
+        # Test non-ascending order buckets
+        metric_family.Metric(labels={}, buckets=[])
```

Comment:
Should this be included? Didn't you find an issue with `buckets=[]`?

Reply:
After the call, I notice something and investigated the error msg again. The mem_leak seems to be the false alarm and not the real error.
The real error was that `metric_family.Metric(labels={}, buckets=[])` successed and created the histogram with empty bucket. Then in the same test I ran the regular histogram e2e test and failed to create a new metric in the same `metric_family`. Not sure if it is expected behavior but I will follow it up.

```
AssertionError: 'test_histogram_e2e_bucket{example1="histogram_label1",example2="histogram_label2",le="0.1"} 0' not found in
# HELP test_histogram_e2e test metric histogram kind end to end
# TYPE test_histogram_e2e histogram
test_histogram_e2e_count{example1="histogram_label1",example2="histogram_label2"} 0
test_histogram_e2e_sum{example1="histogram_label1",example2="histogram_label2"} 0
test_histogram_e2e_bucket{example1="histogram_label1",example2="histogram_label2",le="+Inf"} 0'
```